pep_sphinx_extensions/pep_zero_generator/pep_index_generator.py

```diff
@@ -30,9 +32,25 @@
     from sphinx.environment import BuildEnvironment
 
 
-def create_pep_zero(_: Sphinx, env: BuildEnvironment, docnames: list[str]) -> None:
-    # Sphinx app object is unneeded by this function
+def create_pep_json(peps: list[parser.PEP], out_dir: str) -> None:
+    pep_list = [
+        {
+            "number": pep.number,
+            "title": pep.title,
+            "authors": ", ".join([pep.authors.nick for pep.authors in pep.authors]),
+            "status": pep.status,
+            "type": pep.pep_type,
+            "url": f"https://peps.python.org/pep-{pep.number:0>4}/",
+        }
+        for pep in sorted(peps)
+    ]
 
+    out_file = os.path.join(out_dir, "peps.json")
+    with open(out_file, "w", encoding="UTF-8") as f:
+        json.dump(pep_list, f, indent=0)
```

Comment:
```suggestion
    Path(out_dir, "peps.json").write_text(json.dump(pep_list, f, indent=0), encoding="utf-8")
```

Reply:
Or, equivalently, avoiding an excessively long and complex line,

```suggestion
    peps_json = json.dumps(pep_list, indent=0)
    Path(out_dir, "peps.json").write_text(peps_json, encoding="UTF-8")
```

or perhaps more conventionally, and consistently with the rest of the file,

```suggestion
    with open(Path(out_dir) / "peps.json", "w", encoding="UTF-8") as f:
        json.dump(pep_list, f, indent=0)
```

Reply:
For yet another alternative, there is a `Path.open` method.

Reply:
Thanks! Will return a string and so do none of these :)